archive/qna_engine.py

```
@@ -255,3 +255,70 @@ def get_youtube_transcript(url: str) -> str:
+        results = MCQ_chain.invoke(
+            {"num": num, "topic": topic, **kwargs},
+        )
+        results = results.content
+        structured_output = parser.parse(results)
+
+        llm_math = LLMMathChain.from_llm(llm=llm, verbose=False)
+
+        for question in structured_output.questions:
+            if question.requires_math:
+                try:
+                    with get_openai_callback() as cb:
```

Ruff

`get_openai_callback` may be undefined, or defined from star imports

(F405)